pennylane/data/data_manager/__init__.py

```diff
@@ -423,48 +487,113 @@ def list_attributes(data_name):
+            }
+          }
+        }
+        """,
+        {"input": {"datasetClassId": data_name}},
+    )
 
-def _interactive_request_attributes(options):
+    return response["data"]["datasetClass"]["attributes"]
+
+
+def _interactive_request_data_name(data_names):
+    """Prompt the user to select a data name."""
+    print("Please select the data name from the following:")
+    for i, option in enumerate(data_names):
+        print(f"{i + 1}: {option}")
+    choice = input("Choice of data name: ")
+    if choice not in data_names:
+        raise ValueError(f"Must select a single data name from {data_names}")
+    return choice
+
+
+def _interactive_request_attributes(attribute_options):
     """Prompt the user to select a list of attributes."""
-    prompt = "Please select attributes:"
-    for i, option in enumerate(options):
-        if option == "full":
-            option = "full (all attributes)"
-        prompt += f"\n\t{i+1}) {option}"
-    print(prompt)
-    choices = input(f"Choice (comma-separated list of options) [1-{len(options)}]: ").split(",")
-    try:
-        choices = list(map(int, choices))
-    except ValueError as e:
-        raise ValueError(f"Must enter a list of integers between 1 and {len(options)}") from e
-    if any(choice < 1 or choice > len(options) for choice in choices):
-        raise ValueError(f"Must enter a list of integers between 1 and {len(options)}")
-    return [options[choice - 1] for choice in choices]
-
-
-def _interactive_request_single(node, param):
-    """Prompt the user to select a single option from a list."""
-    options = list(node)
-    if len(options) == 1:
-        print(f"Using {options[0]} as it is the only {param} available.")
-        sleep(1)
-        return options[0]
-    print(f"Please select a {param}:")
-    print("\n".join(f"\t{i+1}) {option}" for i, option in enumerate(options)))
-    try:
-        choice = int(input(f"Choice [1-{len(options)}]: "))
-    except ValueError as e:
-        raise ValueError(f"Must enter an integer between 1 and {len(options)}") from e
-    if choice < 1 or choice > len(options):
-        raise ValueError(f"Must enter an integer between 1 and {len(options)}")
-    return options[choice - 1]
+    print(
+        'Please select a list of attributes from the following available attributes or "full" for all attributes.'
+    )
+    for i, option in enumerate(attribute_options):
+        print(f"{i + 1}: {option}")
+    choice = input("Choice of attributes: ")
+    if choice == "full":
+        return attribute_options
+    if not set(choice).issubset(set(attribute_options)):
+        raise ValueError(f"Must select a list of attributes from {attribute_options}")
+    return choice
+
+
+def _interactive_requests(parameters, parameter_tree):
+    """Prompts the user to select parameters for datasets one at a time."""
+
+    branch = parameter_tree
+    choices = []
+    for param in parameters:
+        print(f"Available options for {param}:")
```

Comment:
1. We should move the leaf check to the top of the loop, this is helpful because it guarantees that `branch` is always a non-leaf node in the rest of the loop
2. A leaf node is a `str`,  not a `dict`

```suggestion
        if isinstance(branch, str):
            break
            
        if len(branch["next"]) == 1:
            branch = next(iter(branch["next"].values()))
            continue
            
        print(f"Available options for {param}:")
```